modules/arm_plugin/src/transformations/convert_pool_arm.cpp

```diff
@@ -44,19 +54,35 @@ ArmPlugin::pass::ConvertArmMaxPoolV8::ConvertArmMaxPoolV8() {
             return false;
         }
 
-        auto arm_pool = std::make_shared<opset::v8::ArmMaxPool>(max_pool->input_value(0),
+        size_t rank = max_pool->get_output_partial_shape(0).size();
+        if (rank < 4 || rank > 5) {
+            return false;
+        }
+        auto axis = max_pool->get_axis();
+        if (axis > 1 || (axis < 0 && axis > -static_cast<int64_t>(rank) - 1)) {
+            return false;
+        }
+        auto activations_transpose = transpose_on_input(max_pool->input_value(0), rank);
+        auto output_shape = transpose_output_shape(max_pool, rank);
+        auto arm_pool = std::make_shared<opset::v8::ArmMaxPool>(activations_transpose,
```

Comment:
is ArmMaxPool class inherited from ov::op::v8::MaxPool?
If so, this conversion could potentially be called twice, I guess we need to check that this is not ArmMaxPool already.
```
auto arm_pool = std::dynamic_pointer_cast<ArmMaxPool>(m.get_match_root());
if (arm_pool)
   return false;
```

Reply:
done